### DISCUSSION TITLE
Nested test() and test() are not executed, with no warning or error message

### DISCUSSION THREAD

**Discussion by mrsekut** [vitest-dev/vitest#3224] [General]
I understand that nesting `test()` functions is not a recommended practice and that they should not be executed. However, I believe it would be beneficial to have a runtime error or a message indicating that the nested `test()` was not executed. Without such a message, users might mistakenly believe that the nested test has passed.

For example, consider the following code:

```javascript
describe('AAA', () => {
  test('aaaa', () => {
    expect(1).toBe(1); // pass. ok
    test('bbb', () => {
      expect(2).toBe(1); // pass?. Not executed in reality
    });
  });
})
```

In this case, the nested `test('bbb', ...)` function is not executed, but there is no warning or error message to inform the user.

This issue is particularly noticeable when using `test.each`, as it is visually easy to accidentally place `test.each` inside another `test()` if they are indented similarly.

Consider another example with `test.each`:

```javascript
test('sum(a, b) returns the correct result', () => {
  const testData = [
    [1, 2, 3],
    [2, 3, 5],
  ];
  test.each(testData)('sum(%i, %i)', (a, b, expected) => {
    expect(sum(a, b)).toBe(expected);  // Not executed in reality
  });
});

test('another test case', () => {
  expect(1).toBe(1); // pass. ok
});
```

In this second example, the `test.each` function is not executed because it is nested inside another `test()`, but there is no warning or error message to inform the user. It would be helpful to provide some feedback to the user when this situation occurs, even though they should not be nesting `test()` functions in the first place.

**Comment by GadgetBlaster** [comment]
I stumble and make this mistake semi-frequently. Having an error thrown, or the ability to lint against this issue, would be a fantastic improvement.

**Comment by sheremet-va** [comment]
Consider opening an enhancement issue. Team members usually filter issues there to see what is needed. I didn't even know there was a discussion about this.

I personally think that it would be a good idea to throw an error in such cases.

**Reply by mrsekut** [reply]
Thanks for the advice. I posted the same information as an issue.
https://github.com/vitest-dev/vitest/issues/4229
